### PR TITLE
add developer documentation

### PR DESCRIPTION
**Pull request by henrykironde**

### CODE REVIEW

**Comment by henrykironde** [comment]
@ethanwhite works fine

File: docs/developer.rst [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+In case you find errors with wxPython, try `installing the wheel`_ from a third party distributor
+
+Setting up servers
+==================
+
+You need to install all the database infrastructures to enable local testing.
+
+::  
+  
+  SQlite
```
Comment (ethanwhite) [inline docs/developer.rst]: SQlite -> SQLite

File: docs/developer.rst [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+In some cases we do tend to work outside the ``pep8`` requirements.
+The compromise on ``pep8``  may be a result of enforcing better code readability.
+In some cases ``pep`` shows errors for long lines, but that can be ignored.
+
+``pep8 pythonfile.py``
+
+Testing
+=======
+
+Follow these instructions to run a complete set of test for any branch
```
Comment (ethanwhite) [inline docs/developer.rst]: "test" -> "tests"

File: docs/developer.rst [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+
+  $ pip setup.py install
+
+or install in development mode.
+
+.. code-block:: bash
+
+  $  pip setup.py develop
+
+For more about `installing`_ refer to the python setuptool `documentation`_.
```
Comment (ethanwhite) [inline docs/developer.rst]: setuptool -> setuptools

File: docs/developer.rst [hunk 4]
```diff
@@ -0,0 +1,204 @@
...
+.. code-block:: sh
+
+  $   py.test
+
+
+In case we want to run tests on a specific test category, we add the path of the test module, py.test [path]
+
+.. code-block:: sh
+
+  $   py.test .\test\test_retriever.py
```
Comment (ethanwhite) [inline docs/developer.rst]: This should probably be `./test/test_retriever.py`. As written it won't run on Linux. I think, but can't remember for sure that Windows will often handle `/` in place of `\`.

File: docs/developer.rst [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+
+This will only run test_retriever.py
+
+Tests in the Cloud
+^^^^^^^^^^^^^^^^^^^^
+
+Currently we support Travis, a cloud testing platform.
+
+In case you make commits, Travis is automatically trigered and uses the ``travis.yml`` file.
```
Comment (ethanwhite) [inline docs/developer.rst]: trigered -> triggered

File: docs/developer.rst [hunk 6]
```diff
@@ -0,0 +1,204 @@
...
+
+Documentation
+=============
+
+We are using `Sphinx`_ and `Read the Docs`_. for the documentation. 
+Sphinx uses reStructuredText as its markup language. 
+Source Code documentation is automatically included after committing to the master.
+Other documentation (not source code) files are added as new reStructuredText in the docs folder
+
+In case you want to change the organization of the Documenntation, please refer to `Sphinx`_
```
Comment (ethanwhite) [inline docs/developer.rst]: Documenntation -> Documentation

File: docs/developer.rst [hunk 7]
```diff
@@ -0,0 +1,204 @@
...
+
+**Test Documentation locally**
+
+.. code-block:: bash
+
+  cd  docs  # go the docs directory 
+  make html # Run
+  
+  Note: 
+  Do not commit the build direcotry after making html.
```
Comment (ethanwhite) [inline docs/developer.rst]: direcotry -> directory

File: docs/developer.rst [hunk 8]
```diff
@@ -0,0 +1,204 @@
...
+ 
+ 
+**Read The Docs  configuration**
```
Comment (ethanwhite) [inline docs/developer.rst]: Extra space between `Docs` and `configuration`

File: docs/developer.rst [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+
+Configure read the docs(advanced settings) so that the source is first installed then docs are built.
```
Comment (ethanwhite) [inline docs/developer.rst]: add space after `docs`

File: docs/developer.rst [hunk 10]
```diff
@@ -0,0 +1,204 @@
...
+
+**Commits**
+
+From your local branch of retriever, commit to your origin.
+Once tests have passed you can then make a pull request to the retriever master (upstream) 
+
+**Clean histroy**
+
+We try to make one commit for each issue.
+As you work on an issue, try adding all the commits into one general commit rather than several commits. you can use commit amend to add new changes to a branch.
```
Comment (ethanwhite) [inline docs/developer.rst]: "you can use commit amend"-> "Use \`git commit --amend\`"

**Comment by ethanwhite** [comment]
Wow @henrykironde! This is really nice developer documentation. Great work!

I noted a few minor fixes in-line. Ping me when they're done and we'll get this in.

**Comment by henrykironde** [comment]
@ethanwhite ready. Please take a final look.

**Comment by ethanwhite** [comment]
Looks great! :+1:

**Pull request merged**